lib/renderer/web-frame-init.js

```diff
@@ -1,38 +1,25 @@
-const electron = require('electron')
+const {ipcRenderer, webFrame} = require('electron')
+const errorUtils = require('../common/error-utils')
 
 module.exports = () => {
   // Call webFrame method
-  electron.ipcRenderer.on('ELECTRON_INTERNAL_RENDERER_WEB_FRAME_METHOD', (event, method, args) => {
-    electron.webFrame[method](...args)
-  })
-
-  electron.ipcRenderer.on('ELECTRON_INTERNAL_RENDERER_SYNC_WEB_FRAME_METHOD', (event, requestId, method, args) => {
-    const result = electron.webFrame[method](...args)
-    event.sender.send(`ELECTRON_INTERNAL_BROWSER_SYNC_WEB_FRAME_RESPONSE_${requestId}`, result)
+  ipcRenderer.on('ELECTRON_INTERNAL_RENDERER_WEB_FRAME_METHOD', (event, method, args) => {
+    try {
+      webFrame[method](...args)
+    } catch (error) {
+      console.error(`${error}`)
```

Comment:
This error shouldn't be just dismissed / logged, it should be catchable by users.

Reply:
@MarshallOfSound to do that I need to make all the methods return a Promise

Reply:
I don't mind if that comes later as part of the larger Promisification project -- but if it's easy to do now, that's even better.

Reply:
@ckerr I would do it later, the main goal of this PR was to move the Error serialization/deserialization code into separate file, as I have follow up PRs depending on that.

Reply:
My point here was before this error was catchable with the global uncaught error handler, now it is not.  Either we make it a promise and catchable here, or we drop this `try {} catch {]`

Reply:
@MarshallOfSound now I get what you meant. I've reverted the try/catch here.